### ISSUE TITLE
Alertmanager configuration customization no longer works

### ISSUE THREAD

**Issue by alexandre-allard** [scality/metalk8s#3134]
**Component**: salt, monitoring

**What happened**:
Trying to update the configuration of Alertmanager to set a receiver using CSC mechanism, it appears that the new configuration is never taken into account.

**What was expected**:
The Alertmanager configuration is updated.

**Steps to reproduce**:
```
$ kubectl edit -n metalk8s-monitoring cm metalk8s-alertmanager-config
# Do whatever you want to Alertmanager's configuration, then apply changes.
$ kubectl exec -n kube-system -c salt-master salt-master-bootstrap -- salt-run state.sls metalk8s.addons.prometheus-operator.deployed saltenv=metalk8s-2.8.0-dev
$ kubectl get -n metalk8s-monitoring secret alertmanager-prometheus-operator-alertmanager -o jsonpath='{ .data.alertmanager\.yaml }' | base64 -d
# Configuration is style the default one
```

**Resolution proposal** (optional):
I don't know in which version it has been introduced, but it seems that the operator is now creating a new secret file `alertmanager-prometheus-operator-alertmanager-generated` which is then used to configure Alertmanager.
The fix is pretty simple, we just have to let the operator knows that we already have a secret.
Basically by adding the following in the chart values and re-rendering it:
```
diff --git a/charts/kube-prometheus-stack.yaml b/charts/kube-prometheus-stack.yaml
index c11690834..2c9a74ec5 100644
--- a/charts/kube-prometheus-stack.yaml
+++ b/charts/kube-prometheus-stack.yaml
@@ -57,6 +57,8 @@ alertmanager:
   # config: '__var_tojson__(alertmanager.spec.notification.config)'

     useExistingSecret: true
+    secrets:
+      - alertmanager-prometheus-operator-alertmanager

 prometheusOperator:
   tls:
```

**Comment by alexandre-allard** [comment]
Fixed by #3136
